src/tasks/tasks.py

```diff
@@ -366,6 +405,76 @@ def get_metrics(self, reset=False):
         return {'mcc': self.scorer1.get_metric(reset),
                 'accuracy': self.scorer2.get_metric(reset)}
 
+@register_task('cola-analysis', rel_path='CoLA/')
+class CoLAAnalysisTask(SingleClassificationTask):
+    def __init__(self, path, max_seq_len, name, **kw):
+        super(CoLAAnalysisTask, self).__init__(name, n_classes=2, **kw)
+        self.load_data(path, max_seq_len)
+        self.sentences = self.train_data_text[0] + self.val_data_text[0]
+        self.val_metric = "%s_mcc" % self.name
+        self.val_metric_decreases = False
+        self.scorer1 = Correlation("matthews")
+        self.scorer2 = CategoricalAccuracy()
+
+    def load_data(self, path, max_seq_len):
+        '''Load the data'''
+        # Load data from tsv
+        tsv_loader = TSVLoader(
+            tokenizer_name=self._tokenizer_name,
+            max_seq_len=max_seq_len,
+            tag_eval=True)
+        
+        tr_data = tsv_loader.load_file(os.path.join(path, "train_analysis.tsv"),
+            s1_cid=3, s2_cid=None, label_cid=2, skip_rows=1, tag2cid_dict={'Domain': 1})
+        val_data = tsv_loader.load_file(os.path.join(path, "dev_analysis.tsv"),
+            s1_cid=3, s2_cid=None, label_cid=2, skip_rows=1, tag2cid_dict={
+                'Domain': 1, 'Simple': 4, 'Pred': 5, 'Adjunct': 6, 'Arg Types': 7,
+                'Arg Altern': 8, 'Imperative': 9, 'Binding': 10, 'Question': 11, 'Comp Clause': 12,
+                'Auxillary': 13, 'to-VP': 14, 'N, Adj': 15, 'S-Syntax': 16, 'Determiner': 17, 'Violations': 18})
```

Comment:
Aren't there more tags than this?

Reply:
Yes, these are only the coarse tags to column index dictionary. I reorganized cola_dev_annotated following the mnli-diagnostic format. If a sample contains multiple fine tags under one coarse tag, the fine tags are listed under corresponding column, separated by ';' .

Reply:
Got it. Does this code read those fine tags? We should.

Reply:
Yes, all the fine tags will be loaded as "coarse__fine". e.g. "Determiner__Partitive"